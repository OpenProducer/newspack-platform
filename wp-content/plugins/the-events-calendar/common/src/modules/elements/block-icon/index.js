/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import { TEC } from '@moderntribe/common/icons';
import './style.pcss';

export default () => (
	<div className="tribe-editor__icons__container tribe-editor__icons--tec">
		<TEC />
	</div>
);
