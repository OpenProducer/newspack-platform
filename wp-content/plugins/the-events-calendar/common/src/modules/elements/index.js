export { default as Accordion } from './accordion/element';
export { default as BlockIcon } from '@moderntribe/common/elements/block-icon';
export { default as Button } from './button/element';

export { default as Counter } from './counter/element';
export { default as Image } from './image/element';
export { default as ImageUpload } from './image-upload/element';
export { default as LabeledItem } from './labeled-item/element';
export { default as LabelWithLink } from './label-with-link/element';
export { default as LabelWithModal } from './label-with-modal/element';
export { default as Link } from './link/element';
export { default as ModalButton } from './modal-button/element';
export { default as TimePicker } from './time-picker/element';
export { default as Tooltip } from './tooltip/element';
export { default as DayPickerInput } from './day-picker-input/element';
export { default as CreatableSelect } from './creatable-select/element';
export { default as Placeholder } from './placeholder/element';
export { default as Heading } from './heading/element';
export { default as Paragraph } from './paragraph/element';

// Inputs
export { default as Input } from './input/element';
export { default as UrlInput } from './url-input/element';
export { default as NumberInput } from './number-input/element';
export { default as Radio } from './radio/element';
export { default as RadioInput } from './radio-input/element';
export { default as Checkbox } from './checkbox/element';
export { default as CheckboxInput } from './checkbox-input/element';
export { default as Select } from './select/element';
export { default as Textarea } from './textarea/element';

/**
 * @todo move this into Editor Module
 */

import './style.pcss'
