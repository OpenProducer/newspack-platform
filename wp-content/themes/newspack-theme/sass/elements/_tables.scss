table {
	margin: 0 0 $size__spacing-unit;
	border-collapse: collapse;
	width: 100%;
	font-size: $font__size-sm;

	td,
	th {
		padding: 0.5em;
		border: 1px solid $color__text-light;
		word-break: break-all;
	}
}
