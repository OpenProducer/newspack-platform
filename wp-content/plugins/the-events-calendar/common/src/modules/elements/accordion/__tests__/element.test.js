/**
 * External dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { noop } from 'lodash';

/**
 * Internal dependencies
 */
import { Accordion } from '@moderntribe/common/elements';

let rows;

describe( 'Accordion Element', () => {
	beforeEach( () => {
		rows = [
			{
				accordionId: '123',
				content: 'this is a content',
				contentClassName: 'content-class',
				header: 'this is header',
				headerClassName: 'header-class',
				onClick: noop,
				onClose: noop,
				onOpen: noop,
			},
		];
	} );

	it( 'renders null', () => {
		const component = renderer.create( <Accordion /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders an accordion', () => {
		const component = renderer.create( <Accordion rows={ rows } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders an accordion with wrapper class', () => {
		const component = renderer.create( <Accordion className="test-class" rows={ rows } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
