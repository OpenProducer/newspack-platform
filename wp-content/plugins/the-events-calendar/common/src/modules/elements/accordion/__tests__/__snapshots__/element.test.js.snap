// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Element renders an accordion 1`] = `
<div
  aria-multiselectable="true"
  className="tribe-editor__accordion"
  role="tablist"
>
  <article
    className="tribe-editor__accordion__row"
  >
    <button
      aria-controls="accordion-content-123"
      aria-expanded="false"
      aria-selected="false"
      className="tribe-editor__button tribe-editor__accordion__row-header header-class"
      id="accordion-header-123"
      onClick={[Function]}
      role="tab"
      type="button"
    >
      this is header
    </button>
    <div
      aria-hidden="true"
      aria-labelledby="accordion-header-123"
      className="tribe-editor__accordion__row-content content-class"
      id="accordion-content-123"
      role="tabpanel"
    >
      this is a content
    </div>
  </article>
</div>
`;

exports[`Accordion Element renders an accordion with wrapper class 1`] = `
<div
  aria-multiselectable="true"
  className="tribe-editor__accordion test-class"
  role="tablist"
>
  <article
    className="tribe-editor__accordion__row"
  >
    <button
      aria-controls="accordion-content-123"
      aria-expanded="false"
      aria-selected="false"
      className="tribe-editor__button tribe-editor__accordion__row-header header-class"
      id="accordion-header-123"
      onClick={[Function]}
      role="tab"
      type="button"
    >
      this is header
    </button>
    <div
      aria-hidden="true"
      aria-labelledby="accordion-header-123"
      className="tribe-editor__accordion__row-content content-class"
      id="accordion-content-123"
      role="tabpanel"
    >
      this is a content
    </div>
  </article>
</div>
`;

exports[`Accordion Element renders null 1`] = `null`;
